### PR TITLE
FOUNDATIONS: Profile Remove By Id

### LINKED ISSUE #203: FOUNDATIONS: Remove Group
@hassanhabib since the Model and Broker is already in place can I continue with this bro?

### PR DESCRIPTION
**Pull request by rcruzfreelance**

### CODE REVIEW

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+using Taarafo.Core.Models.Profiles;
+using Taarafo.Core.Models.Profiles.Exceptions;
+using Xunit;
+
+namespace Taarafo.Core.Tests.Unit.Services.Foundations.Profiles
+{
+    public partial class ProfileServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowDependencyValidationOnRemoveIfDatabaseUpdateConcurrencyErrorOccursAndLogItAsync()
```
Comment (ShriHumrudha) [inline Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs:20]: First test should be Sql Exception.

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                broker.DeleteProfileAsync(It.IsAny<Profile>()),
+                    Times.Never);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowDependencyExceptionOnDeleteWhenSqlExceptionOccursAndLogItAsync()
```
Comment (ShriHumrudha) [inline Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs:65]: Title to have 'Critical'

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+            var expectedProfileDependencyException =
+                new ProfileDependencyException(failedProfileStorageException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectProfileByIdAsync(It.IsAny<Guid>()))
+                    .ThrowsAsync(sqlException);
+
+            // when
+            ValueTask<Profile> deleteProfileTask =
```
Comment (ShriHumrudha) [inline Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs:82]: RemoveProfileByIdTask

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectProfileByIdAsync(It.IsAny<Guid>()),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogCritical(It.Is(SameExceptionAs(
+                    expectedProfileDependencyException))),
+                        Times.Once);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
```
Comment (ShriHumrudha) [inline Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs:98]: Verify Remove Never like before

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectProfileByIdAsync(It.IsAny<Guid>()),
+                        Times.Once());
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedProfileServiceException))),
+                        Times.Once);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
```
Comment (ShriHumrudha) [inline Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs:137]: Delete never verification is missing

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Logic.RemoveById.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            // when
+            Profile actualProfile = await this.profileService
+                .RemoveProfileByIdAsync(inputProfileId);
+
+            // then
+            actualProfile.Should().BeEquivalentTo(expectedProfile);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectProfileByIdAsync(inputProfileId),
+                Times.Once());
```
Comment (ShriHumrudha) [inline Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Logic.RemoveById.cs:46]: Tab here

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Validations.RemoveById.cs [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+            await Assert.ThrowsAsync<ProfileValidationException>(() =>
+                removeProfileByIdTask.AsTask());
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedProfileValidationException))),
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectProfileByIdAsync(invalidProfileId),
```
Comment (ShriHumrudha) [inline Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Validations.RemoveById.cs:47]: It.IsAny

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Validations.RemoveById.cs [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+                    Times.Never);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.DeleteProfileAsync(It.IsAny<Profile>()),
+                    Times.Never);
+        }
+
+        [Fact]
+        public async Task ShouldThrowNotFoundExceptionOnRemoveIdProfileIsNotFounfAndLogItAsync()
```
Comment (ShriHumrudha) [inline Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Validations.RemoveById.cs:56]: Either 'OnRemove' or 'OnRemoveById'

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Validations.RemoveById.cs [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+            // then
+            await Assert.ThrowsAsync<ProfileValidationException>(() =>
+                removeProfileByIdTask.AsTask());
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectProfileByIdAsync(It.IsAny<Guid>()),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(expectedProfileValidationException))),
```
Comment (ShriHumrudha) [inline Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Validations.RemoveById.cs:85]: Consider breaking it to two lines like other tests

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Validations.RemoveById.cs [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+                    Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.DeleteProfileAsync(It.IsAny<Profile>()),
+                    Times.Never);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+
```
Comment (ShriHumrudha) [inline Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Validations.RemoveById.cs:95]: Remove extra line

File: Taarafo.Core/Services/Foundations/Profiles/IProfileService.cs [hunk 11]
```diff
@@ -16,5 +16,7 @@ public partial interface IProfileService
         ValueTask<Profile> RetrieveProfileByIdAsync(Guid profileId);
         IQueryable<Profile> RetrieveAllProfiles();
         ValueTask<Profile> ModifyProfileAsync(Profile profile);
+        ValueTask<Profile> RemoveProfileByIdAsync(Guid profileId);
+
```
Comment (ShriHumrudha) [inline Taarafo.Core/Services/Foundations/Profiles/IProfileService.cs:20]: Remove extra line

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs (hunk 1 above)
Comment (rcruzfreelance) [reply Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs:20]: done

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs (hunk 2 above)
Comment (rcruzfreelance) [reply Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs:65]: done

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs (hunk 3 above)
Comment (rcruzfreelance) [reply Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs:82]: done

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs (hunk 4 above)
Comment (rcruzfreelance) [reply Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs:98]: done

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs (hunk 5 above)
Comment (rcruzfreelance) [reply Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs:137]: done

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Validations.RemoveById.cs (hunk 7 above)
Comment (rcruzfreelance) [reply Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Validations.RemoveById.cs:47]: done

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Validations.RemoveById.cs (hunk 8 above)
Comment (rcruzfreelance) [reply Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Validations.RemoveById.cs:56]: done

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Validations.RemoveById.cs (hunk 9 above)
Comment (rcruzfreelance) [reply Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Validations.RemoveById.cs:85]: done

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Validations.RemoveById.cs (hunk 10 above)
Comment (rcruzfreelance) [reply Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Validations.RemoveById.cs:95]: done

File: Taarafo.Core/Services/Foundations/Profiles/IProfileService.cs (hunk 11 above)
Comment (rcruzfreelance) [reply Taarafo.Core/Services/Foundations/Profiles/IProfileService.cs:20]: done

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Logic.RemoveById.cs (hunk 6 above)
Comment (rcruzfreelance) [reply Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Logic.RemoveById.cs:46]: done

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+using Taarafo.Core.Models.Profiles;
+using Taarafo.Core.Models.Profiles.Exceptions;
+using Xunit;
+
+namespace Taarafo.Core.Tests.Unit.Services.Foundations.Profiles
+{
+    public partial class ProfileServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowCriticalDependencyExceptionOnDeleteWhenSqlExceptionOccursAndLogItAsync()
```
Comment (ShriHumrudha) [inline Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs:20]: OnRemove

File: Taarafo.Core/Services/Foundations/Profiles/IProfileService.cs (hunk 11 above)
Comment (ShriHumrudha) [reply Taarafo.Core/Services/Foundations/Profiles/IProfileService.cs:20]: I still see line 20

image

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

**Comment by ShriHumrudha** [comment]
Closing due to inactivity.

**Comment by rcruzfreelance** [comment]
Could you please reopen this I had no Power for a while and I just got it back

Sent from my T-Mobile 4G LTE Device
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

File: Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs (hunk 12 above)
Comment (rcruzfreelance) [reply Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RemoveById.cs:20]: done

**Review by ShriHumrudha — APPROVED** [review]

**Pull request merged**